main.go

```diff
@@ -48,21 +50,21 @@ func main() {
 		return
 	}
 
-	err = validation.Validate(opts)
-	if err != nil {
-		logger.Printf("%s", err)
-		os.Exit(1)
+	if err = validation.Validate(opts); err != nil {
+		logger.Fatalf("%s", err)
 	}
 
 	validator := NewValidator(opts.EmailDomains, opts.AuthenticatedEmailsFile)
 	oauthproxy, err := NewOAuthProxy(opts, validator)
 	if err != nil {
-		logger.Errorf("ERROR: Failed to initialise OAuth2 Proxy: %v", err)
-		os.Exit(1)
+		logger.Fatalf("ERROR: Failed to initialise OAuth2 Proxy: %v", err)
 	}
 
 	rand.Seed(time.Now().UnixNano())
 
+	m := &http.Server{Addr: opts.MetricsPort, Handler: middleware.DefaultMetricsHandler}
+	m.ListenAndServe()
```

Comment:
IIRC this is a blocking call, it will need to be in a go routine right?

Reply:
Yes, it will, oops. :facepalm: